### ISSUE TITLE
[librsvg] update to <2.54.1>

### ISSUE THREAD

**Issue by Baklap4** [microsoft/vcpkg#24402]
Library name: Librsvg

New version number: [2.54.1](https://gitlab.gnome.org/GNOME/librsvg/-/tags/2.54.1)

The current release 2.40.x is pretty outdated compared to the latest.
It'd be nice to have support for a more recent version, latest being 2.54.1 as of writing

**Comment by FrankXie05** [comment]
@Baklap4 
Currently, due to port `librsvg` must be built with `rust`. but It looks like rustup doesn't provide a full tarball, but uses the built-in management tools to install and manage its dependencies.
So we can't easily add it as a tool, we can only follow `vcpkg_acquire_msys`.
We may not update liibrsvg anytime soon. :(

**Comment by Lord-Kamina** [comment]
> @Baklap4 Currently, due to port `librsvg` must be built with `rust`. but It looks like rustup doesn't provide a full tarball, but uses the built-in management tools to install and manage its dependencies. So we can't easily add it as a tool, we can only follow `vcpkg_acquire_msys`. We may not update liibrsvg anytime soon. :(

I am part of the project @Baklap4 is working on using librsvg. A while ago, I made a librsvg port for mxe (cross-compiler using mingw-w64) and what I learned is, you can download and install ready-made rustc binaries and the rust stdlib, and use those to compile librsvg.

Here are the relevant bits of code I used. I think if you were to use this approach, it would not only be doable, but probably much simpler than the hoops I had to go through considering I was cross-compiling.

TL;DR of what's below is: You can install canned binaries for all the rust components, and use those to compile what we need.


First, I installed a "bootstrap" rust stdlib, as defined here, (note all those checksums were to acommodate the different possible host machines, if you're not gonna cross-compile, you can probably simplify or at least divide them per-OS) 

https://github.com/Lord-Kamina/mxe/blob/performous/src/rust-std-bootstrap.mk

I then installed the rust compiler binary like this (again, note that all those checksums are because I wanted to acommodate different possible host machies): https://github.com/Lord-Kamina/mxe/blob/performous/src/rustc.mk

Then I install the rust-std lib (notice here how I was providing checksums only for the i386 and x86_64 windows versions): https://github.com/Lord-Kamina/mxe/blob/performous/src/rust-std.mk

Then install cargo: https://github.com/Lord-Kamina/mxe/blob/performous/src/cargo.mk

And finally, actually compile librsvg: https://github.com/Lord-Kamina/mxe/blob/performous/src/librsvg.mk

**Comment by FrankXie05** [comment]
@Lord-Kamina Yes, if I install rust locally, it can work . But the problem is that we cannot force users to install any software on the system level. We can consider integrating on vcpkg.

**Comment by Baklap4** [comment]
Would be nice if this is going to work 👍

**Comment by FrankXie05** [comment]
Wait https://github.com/microsoft/vcpkg/pull/30251

**Comment by luzpaz** [comment]
Meanwhile 2.59.2 is current stable (https://gitlab.gnome.org/GNOME/librsvg/-/releases/2.59.2)

**Comment by talregev** [comment]
Also needed for compile gimp.
Any other suggestions?
